### ISSUE TITLE
Fallo en Set Files as default file explorer

### ISSUE THREAD

**Issue by AitorPro** [files-community/Files#7308]
### Description

No puedo desactivar la opcion de Set Files as default file explorer y no me funciona la opcion

Espero que lo solucionen en una proxima version

Me despido 

### Steps To Reproduce

Voy a opciones y voy a experimental y activo esta opcion Set Files as default file explorer

### Expected behavior

Hasta luego

### Files Version

Versión: 2.0.34.0 OS Version: 10.0.22523.1000

### Windows Version

Windows 11 Pro Insider Preview Versión Build 22523.1000

### Relevant Assets/Logs

debug.log

2021-12-16 21:54:39.2491|INFO|OnLaunched|App launched. Prelaunch: False
2021-12-16 21:54:39.9209|INFO|GetDrivesAsync|Drive added: C:\, Fixed
2021-12-16 21:54:43.2174|INFO|EnumerateDrivesAsync|Adding cloud provider "Google Drive (Aitor)" mapped to C:\Users\Aitor\OneDrive\Aitor
2021-12-16 21:55:02.6511|INFO|OnActivated|App activated by CommandLineLaunch
2021-12-16 21:55:14.8503|INFO|IsRunningOnArm|Running on ARM: False
2021-12-16 22:03:45.5774|INFO|OnActivated|App activated by CommandLineLaunch
2021-12-16 22:03:46.0773|INFO|GetDrivesAsync|Drive added: C:\, Fixed
2021-12-16 22:03:46.9054|INFO|EnumerateDrivesAsync|Adding cloud provider "Google Drive (Aitor)" mapped to C:\Users\Aitor\OneDrive\Aitor
2021-12-16 22:05:09.2731|INFO|OnLaunched|App launched. Prelaunch: False
2021-12-16 22:05:09.7261|INFO|GetDrivesAsync|Drive added: C:\, Fixed
2021-12-16 22:05:10.5854|INFO|EnumerateDrivesAsync|Adding cloud provider "Google Drive (Aitor)" mapped to C:\Users\Aitor\OneDrive\Aitor
2021-12-16 22:05:36.6677|INFO|OnActivated|App activated by CommandLineLaunch
2021-12-16 22:05:37.2457|INFO|GetDrivesAsync|Drive added: C:\, Fixed
2021-12-16 22:05:38.2300|WARN|SendMessageForResponseAsync|Error sending request on pipe.
	System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0xad
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17
   at Files.Helpers.NamedPipeAsAppServiceConnection.<SendMessageForResponseAsync>d__12.MoveNext() + 0x24d

2021-12-16 22:05:38.2612|WARN|SendMessageForResponseAsync|Error sending request on pipe.
	System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0xad
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17
   at Files.Helpers.NamedPipeAsAppServiceConnection.<SendMessageForResponseAsync>d__12.MoveNext() + 0x24d

2021-12-16 22:05:38.2769|WARN|SendMessageForResponseAsync|Error sending request on pipe.
	System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0xad
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17
   at Files.Helpers.NamedPipeAsAppServiceConnection.<SendMessageForResponseAsync>d__12.MoveNext() + 0x24d

2021-12-16 22:05:38.2925|WARN|SendMessageForResponseAsync|Error sending request on pipe.
	System.Threading.Tasks.TaskCanceledException: A task was canceled.
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0xad
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17
   at Files.Helpers.NamedPipeAsAppServiceConnection.<SendMessageForResponseAsync>d__12.MoveNext() + 0x24d

2021-12-16 22:05:38.3394|INFO|EnumerateDrivesAsync|Adding cloud provider "Google Drive (Aitor)" mapped to C:\Users\Aitor\OneDrive\Aitor
2021-12-16 22:05:54.5147|INFO|OnLaunched|App launched. Prelaunch: False
2021-12-16 22:05:55.7697|INFO|GetDrivesAsync|Drive added: C:\, Fixed
2021-12-16 22:05:58.1107|INFO|EnumerateDrivesAsync|Adding cloud provider "Google Drive (Aitor)" mapped to C:\Users\Aitor\OneDrive\Aitor
2021-12-16 22:08:30.8816|INFO|OnActivated|App activated by CommandLineLaunch
2021-12-16 22:08:31.6821|INFO|IsRunningOnArm|Running on ARM: False

debug_fulltrust.log

2021-12-16 21:54:42.1394|INFO|Connection_RequestReceived|Argument: GetNewContextMenuEntries
2021-12-16 21:54:42.2956|INFO|Connection_RequestReceived|Argument: ShellLibrary
2021-12-16 21:54:42.3112|INFO|Connection_RequestReceived|Argument: DetectCloudDrives
2021-12-16 21:54:43.2018|INFO|Connection_RequestReceived|Argument: NetworkDriveOperation
2021-12-16 21:54:43.2174|INFO|Connection_RequestReceived|Argument: GetSelectedIconsFromDLL
2021-12-16 21:54:43.2486|INFO|Connection_RequestReceived|Argument: GetIconWithoutOverlay
2021-12-16 21:54:43.2799|INFO|Connection_RequestReceived|Argument: GetSelectedIconsFromDLL
2021-12-16 21:54:43.4049|INFO|Connection_RequestReceived|Argument: DetectQuickLook
2021-12-16 21:54:43.9049|INFO|CheckQuickLookAvailability|QuickLook detected: False
2021-12-16 21:54:44.0142|INFO|Connection_RequestReceived|Argument: SetAsDefaultExplorer
2021-12-16 21:54:47.7043|INFO|Connection_RequestReceived|Argument: SetAsDefaultExplorer
2021-12-16 21:54:49.9646|INFO|Connection_RequestReceived|Argument: SetAsDefaultExplorer
2021-12-16 21:54:59.2694|INFO|Connection_RequestReceived|Argument: SetAsDefaultExplorer
2021-12-16 21:55:07.1194|INFO|Connection_RequestReceived|Argument: SetAsDefaultExplorer
2021-12-16 21:55:15.0065|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 21:55:15.0846|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 21:55:15.0846|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 21:55:15.1002|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 21:55:21.2091|INFO|Connection_RequestReceived|Argument: SetAsDefaultExplorer
2021-12-16 22:00:07.7142|INFO|Connection_RequestReceived|Argument: Terminate
2021-12-16 22:03:46.0930|INFO|Connection_RequestReceived|Argument: GetNewContextMenuEntries
2021-12-16 22:03:46.2492|INFO|Connection_RequestReceived|Argument: DetectCloudDrives
2021-12-16 22:03:46.9054|INFO|Connection_RequestReceived|Argument: NetworkDriveOperation
2021-12-16 22:03:46.9054|INFO|Connection_RequestReceived|Argument: ShellLibrary
2021-12-16 22:03:46.9210|INFO|Connection_RequestReceived|Argument: GetSelectedIconsFromDLL
2021-12-16 22:03:46.9522|INFO|Connection_RequestReceived|Argument: GetIconWithoutOverlay
2021-12-16 22:03:46.9679|INFO|Connection_RequestReceived|Argument: GetSelectedIconsFromDLL
2021-12-16 22:03:47.0773|INFO|Connection_RequestReceived|Argument: DetectQuickLook
2021-12-16 22:03:47.5773|INFO|CheckQuickLookAvailability|QuickLook detected: False
2021-12-16 22:03:48.3896|INFO|Connection_RequestReceived|Argument: Terminate
2021-12-16 22:05:09.8043|INFO|Connection_RequestReceived|Argument: GetNewContextMenuEntries
2021-12-16 22:05:09.9136|INFO|Connection_RequestReceived|Argument: DetectCloudDrives
2021-12-16 22:05:10.5854|INFO|Connection_RequestReceived|Argument: NetworkDriveOperation
2021-12-16 22:05:10.6167|INFO|Connection_RequestReceived|Argument: ShellLibrary
2021-12-16 22:05:10.6167|INFO|Connection_RequestReceived|Argument: GetSelectedIconsFromDLL
2021-12-16 22:05:10.6323|INFO|Connection_RequestReceived|Argument: GetIconWithoutOverlay
2021-12-16 22:05:10.6479|INFO|Connection_RequestReceived|Argument: GetSelectedIconsFromDLL
2021-12-16 22:05:10.7573|INFO|Connection_RequestReceived|Argument: DetectQuickLook
2021-12-16 22:05:11.2572|INFO|CheckQuickLookAvailability|QuickLook detected: False
2021-12-16 22:05:16.3350|INFO|Connection_RequestReceived|Argument: SetAsDefaultExplorer
2021-12-16 22:05:22.3520|INFO|Connection_RequestReceived|Argument: SetAsDefaultExplorer
2021-12-16 22:05:25.6458|INFO|Connection_RequestReceived|Argument: Terminate
2021-12-16 22:05:37.2457|INFO|Connection_RequestReceived|Argument: GetNewContextMenuEntries
2021-12-16 22:05:37.4020|INFO|Connection_RequestReceived|Argument: DetectCloudDrives
2021-12-16 22:05:38.3862|ERROR|UnhandledExceptionTrapper|Canalización interrumpida.
	System.IO.IOException: Canalización interrumpida.
   en System.IO.Pipes.NamedPipeClientStream.CheckPipePropertyOperations()
   en System.IO.Pipes.PipeStream.get_InBufferSize()
   en FilesFullTrust.Program.EndReadCallBack(IAsyncResult result)
   en System.IO.Pipes.PipeStream.AsyncPSCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* pOverlapped)
   en System.Threading._IOCompletionCallback.PerformIOCompletionCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* pOVERLAP)
2021-12-16 22:05:55.8267|INFO|Connection_RequestReceived|Argument: GetNewContextMenuEntries
2021-12-16 22:05:56.1587|INFO|Connection_RequestReceived|Argument: DetectCloudDrives
2021-12-16 22:05:58.1107|INFO|Connection_RequestReceived|Argument: NetworkDriveOperation
2021-12-16 22:05:58.1197|INFO|Connection_RequestReceived|Argument: ShellLibrary
2021-12-16 22:05:58.1447|INFO|Connection_RequestReceived|Argument: GetSelectedIconsFromDLL
2021-12-16 22:05:58.2077|INFO|Connection_RequestReceived|Argument: GetIconWithoutOverlay
2021-12-16 22:05:58.2478|INFO|Connection_RequestReceived|Argument: GetSelectedIconsFromDLL
2021-12-16 22:05:58.5528|INFO|Connection_RequestReceived|Argument: DetectQuickLook
2021-12-16 22:05:59.0448|INFO|CheckQuickLookAvailability|QuickLook detected: False
2021-12-16 22:06:01.6357|INFO|Connection_RequestReceived|Argument: Terminate
2021-12-16 22:06:16.4405|INFO|Connection_RequestReceived|Argument: SetAsDefaultExplorer
2021-12-16 22:07:04.5282|INFO|Connection_RequestReceived|Argument: SetAsDefaultExplorer
2021-12-16 22:08:31.8227|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 22:08:31.9790|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 22:08:31.9946|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 22:08:31.9946|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 22:08:32.0102|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 22:08:32.0102|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 22:08:32.0259|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 22:08:32.0571|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 22:08:32.0727|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 22:15:53.7900|INFO|Connection_RequestReceived|Argument: ShellLibrary
2021-12-16 22:15:53.8308|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 22:15:53.8449|INFO|StartSTATask|El sistema no puede encontrar el archivo especificado. (Excepción de HRESULT: 0x80070002)
	System.IO.FileNotFoundException: El sistema no puede encontrar el archivo especificado. (Excepción de HRESULT: 0x80070002)
   en Vanara.PInvoke.Shell32.ShellUtil.GetShellItemForPath(String path)
   en FilesFullTrust.Win32API.GetFileIconAndOverlay(String path, Int32 thumbnailSize, Boolean getOverlay, Boolean onlyGetOverlay)
   en FilesFullTrust.MessageHandlers.Win32MessageHandler.<>c__DisplayClass3_0.<ParseArgumentsAsync>b__0()
   en FilesFullTrust.Win32API.<>c__DisplayClass0_0`1.<StartSTATask>b__0()
2021-12-16 22:15:53.9380|INFO|Connection_RequestReceived|Argument: GetIconWithoutOverlay
2021-12-16 22:15:53.9717|INFO|Connection_RequestReceived|Argument: GetIconWithoutOverlay
2021-12-16 22:15:53.9858|INFO|Connection_RequestReceived|Argument: GetIconWithoutOverlay
2021-12-16 22:15:54.0002|INFO|Connection_RequestReceived|Argument: GetIconWithoutOverlay
2021-12-16 22:15:57.3764|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 22:15:57.3813|INFO|StartSTATask|El sistema no puede encontrar el archivo especificado. (Excepción de HRESULT: 0x80070002)
	System.IO.FileNotFoundException: El sistema no puede encontrar el archivo especificado. (Excepción de HRESULT: 0x80070002)
   en Vanara.PInvoke.Shell32.ShellUtil.GetShellItemForPath(String path)
   en FilesFullTrust.Win32API.GetFileIconAndOverlay(String path, Int32 thumbnailSize, Boolean getOverlay, Boolean onlyGetOverlay)
   en FilesFullTrust.MessageHandlers.Win32MessageHandler.<>c__DisplayClass3_0.<ParseArgumentsAsync>b__0()
   en FilesFullTrust.Win32API.<>c__DisplayClass0_0`1.<StartSTATask>b__0()
2021-12-16 22:15:59.6783|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 22:15:59.6828|INFO|StartSTATask|El sistema no puede encontrar el archivo especificado. (Excepción de HRESULT: 0x80070002)
	System.IO.FileNotFoundException: El sistema no puede encontrar el archivo especificado. (Excepción de HRESULT: 0x80070002)
   en Vanara.PInvoke.Shell32.ShellUtil.GetShellItemForPath(String path)
   en FilesFullTrust.Win32API.GetFileIconAndOverlay(String path, Int32 thumbnailSize, Boolean getOverlay, Boolean onlyGetOverlay)
   en FilesFullTrust.MessageHandlers.Win32MessageHandler.<>c__DisplayClass3_0.<ParseArgumentsAsync>b__0()
   en FilesFullTrust.Win32API.<>c__DisplayClass0_0`1.<StartSTATask>b__0()
2021-12-16 22:16:00.5589|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 22:16:00.5629|INFO|StartSTATask|El sistema no puede encontrar el archivo especificado. (Excepción de HRESULT: 0x80070002)
	System.IO.FileNotFoundException: El sistema no puede encontrar el archivo especificado. (Excepción de HRESULT: 0x80070002)
   en Vanara.PInvoke.Shell32.ShellUtil.GetShellItemForPath(String path)
   en FilesFullTrust.Win32API.GetFileIconAndOverlay(String path, Int32 thumbnailSize, Boolean getOverlay, Boolean onlyGetOverlay)
   en FilesFullTrust.MessageHandlers.Win32MessageHandler.<>c__DisplayClass3_0.<ParseArgumentsAsync>b__0()
   en FilesFullTrust.Win32API.<>c__DisplayClass0_0`1.<StartSTATask>b__0()
2021-12-16 22:16:02.3611|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 22:16:02.3661|INFO|StartSTATask|El sistema no puede encontrar el archivo especificado. (Excepción de HRESULT: 0x80070002)
	System.IO.FileNotFoundException: El sistema no puede encontrar el archivo especificado. (Excepción de HRESULT: 0x80070002)
   en Vanara.PInvoke.Shell32.ShellUtil.GetShellItemForPath(String path)
   en FilesFullTrust.Win32API.GetFileIconAndOverlay(String path, Int32 thumbnailSize, Boolean getOverlay, Boolean onlyGetOverlay)
   en FilesFullTrust.MessageHandlers.Win32MessageHandler.<>c__DisplayClass3_0.<ParseArgumentsAsync>b__0()
   en FilesFullTrust.Win32API.<>c__DisplayClass0_0`1.<StartSTATask>b__0()
2021-12-16 22:16:03.0784|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 22:16:06.9143|INFO|Connection_RequestReceived|Argument: GetIconOverlay
2021-12-16 22:16:06.9176|INFO|StartSTATask|El sistema no puede encontrar el archivo especificado. (Excepción de HRESULT: 0x80070002)
	System.IO.FileNotFoundException: El sistema no puede encontrar el archivo especificado. (Excepción de HRESULT: 0x80070002)
   en Vanara.PInvoke.Shell32.ShellUtil.GetShellItemForPath(String path)
   en FilesFullTrust.Win32API.GetFileIconAndOverlay(String path, Int32 thumbnailSize, Boolean getOverlay, Boolean onlyGetOverlay)
   en FilesFullTrust.MessageHandlers.Win32MessageHandler.<>c__DisplayClass3_0.<ParseArgumentsAsync>b__0()
   en FilesFullTrust.Win32API.<>c__DisplayClass0_0`1.<StartSTATask>b__0()
2021-12-16 22:16:11.4970|INFO|Connection_RequestReceived|Argument: InvokeVerb
2021-12-16 22:16:48.4445|INFO|Connection_RequestReceived|Argument: Terminate

**Comment by BanCrash** [comment]
Hi @AitorPro  , you have to follow the instructions here to unset as default file explorer:

https://files.community/docs/configuring/replace-file-explorer

TL;DR: download [this zip](https://raw.githubusercontent.com/files-community/files-community.github.io/main/data/SetFilesAsDefault.zip) and run UnsetFilesAsDefault.reg to restore windows explorer
